src/background-view.cpp

```diff
@@ -0,0 +1,355 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 Scott Moreau
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <map>
+#include <fcntl.h>
+#include <unistd.h>
+#include <signal.h>
+#include <sys/wait.h>
+#include <sys/socket.h>
+#include "wayfire/core.hpp"
+#include "wayfire/view.hpp"
+#include "wayfire/plugin.hpp"
+#include "wayfire/output.hpp"
+#include "wayfire/signal-definitions.hpp"
+#include "wayfire/workspace-manager.hpp"
+#include "wayfire/output-layout.hpp"
+
+#include <wayfire/util/log.hpp>
+
+extern "C"
+{
+#include <wlr/config.h>
+#if WLR_HAS_XWAYLAND
+#define class class_t
+#define static
+#include <wlr/xwayland.h>
+#undef static
+#undef class
+#endif
+}
+
+struct process
+{
+    nonstd::observer_ptr<wf::view_interface_t> view;
+    struct wl_client *client;
+    pid_t pid;
+};
+
+static std::map<wf::output_t *, struct process> procs;
+
+class wayfire_background_view : public wf::plugin_interface_t
+{
+    const std::string transformer_name = "background-view";
+    wf::config::option_base_t::updated_callback_t option_changed;
+    wf::option_wrapper_t<std::string> command{"background-view/command"};
+    wf::option_wrapper_t<std::string> file{"background-view/file"};
+    struct wl_event_source *signal = nullptr;
+
+    public:
+    void init() override
+    {
+        grab_interface->name = transformer_name;
+        grab_interface->capabilities = 0;
+
+        option_changed = [=] ()
```

Comment:
Move this outside of the constructor.

Reply:
I assume you mean out of the init function.